### PR TITLE
ui: Sortable issues table and some cleanup&refactoring for tables

### PR DESCRIPTION
**Pull request by Etsija**
Please have a look at the commits for details.

### CODE REVIEW

File: ui/src/helpers/sorting-functions.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * @param b The second rating to compare.
+ * @returns A negative value if `a` is less severe than `b`, a positive value if `a` is more severe than `b`, or zero if
+ *          they are equally severe.
+ */
+export function sortByVulnerabilityRating(
+  a: VulnerabilityRating,
+  b: VulnerabilityRating
+): number {
+  return VulnerabilityRatings[a] - VulnerabilityRatings[b];
+}
```
Comment (mmurto) [inline ui/src/helpers/sorting-functions.ts:65]: IMO thse are more of comparison functions than sorting functions, so I'd rename them to `compareSeverity` or something like that.

File: ui/src/helpers/get-status-class.ts [hunk 2]
```diff
@@ -31,6 +31,8 @@ export const VulnerabilityRatings = {
   NONE: 0,
 } as const;
 
+export type VulnerabilityRating = keyof typeof VulnerabilityRatings;
```
Comment (mmurto) [inline ui/src/helpers/get-status-class.ts:34]: As (I believe) this type is now used where type is needed, I'd rename `VulnerabilityRatings` to `vulnerabilityRatings` to follow conventions more closely.

File: ui/src/helpers/get-package-id.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import { Identifier } from '@/api/requests';
+
+export function getPackageId(pkg: Identifier | undefined | null): string {
+  if (!pkg) {
+    return '';
+  }
+  const { type, namespace, name, version } = pkg;
+  return `${type ? type.concat(':') : ''}${namespace ? namespace.concat('/') : ''}${name ? name : ''}${version ? '@'.concat(version) : ''}`;
+}
```
Comment (mmurto) [inline ui/src/helpers/get-package-id.ts:28]: A more descriptive name for this could be `identifierToString`.

**Review by mmurto — CHANGES_REQUESTED** [review]

File: ui/src/helpers/get-status-class.ts (hunk 2 above)
Comment (Etsija) [reply ui/src/helpers/get-status-class.ts:34]: No, sry, I misunderstood your comment. Will fix.

**Comment by Etsija** [comment]
Due to some severe malfunctioning of GitButler, I need to close this PR and open up a new one, with all change requests addressed.

**Comment by sschuberth** [comment]
@Etsija please remember to delete your `jk-branch-2` branch if no longer needed.

**Pull request closed without merging**